trucks.js

```diff
@@ -1,504 +1,577 @@
+      schedule: ['Tuesday', 'Wednesday', 'Thursday', 'Friday'],
+      payment: ['Cash', 'Cards'],
+      description: 'Modern Middle eastern street food (and drink)!',
+      Facebook: 'https://www.facebook.com/FezOnWheels/info',
+      Twitter: 'https://twitter.com/fezonwheels'
+    },
+    {
+      name: 'Flair Taco',
+      type: ['Mexican'],
+      schedule: ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday'],
+      payment: ['Cash'],
+      description: 'Regular taco truck in the Fremont neighborhood - open 7 days a week and late night',
+      Facebook: 'https://www.facebook.com/pages/Flair-Taco/197832600252403'
+    },
+    {
+      name: 'Frank\'s Franks',
+      type: ['Hot Dogs'],
+      schedule: ['Monday', 'Sunday'],
+      payment: ['Cash', 'Cards'],
+      description: 'Gourmet hot dogs, weiners, sausages, hot links and more, contact at 425.761.9447 or franksfranks3@gmail.com',
+      Facebook: 'https://www.facebook.com/franksfrankshotdogs'
+    },
+    {
+      name: 'Fruit Chatter Box',
+      type: ['Sweets'],
+      schedule: ['Sunday'],
+      payment: ['Cash', 'Cards'],
+      description: 'Fruit Chatter Box provides fresh, seasonal fresh fruit in cups or on a stick at local festivals and street fairs.',
+      Facebook: 'https://www.facebook.com/fruitchatterbox',
+      Twitter: 'https://twitter.com/fruitchatterbox'
+    },
+    {
+      name: 'Full Tilt Ice Cream',
+      type: ['Sweets'],
+      payment: ['Cash', 'Cards'],
+      description: 'Handmade ice creams!',
+      website: 'http://www.fulltilticecream.com',
+      Facebook: 'https://www.facebook.com/fulltilticecreams',
+      Twitter: 'https://twitter.com/fticecream'
+    }
+  ];
+  
+  function  filterTrucksByDay(day) {
+    return (foodTrucks.filter(function(item){
+      var scheduleLen = item.schedule.length;
+      for (var i=0; i<scheduleLen; i++) {
+        if(item.schedule[i] == day) {
+          return true;
+        }
+      }
+    }));
   }
-];
\ No newline at end of file
+
+  function getTrucks(){
+    return foodTrucks;
+  }
+
+  function getTruck(name) {
+    return (foodTrucks.filter(function(item){
+      return (item.name.toUpperCase() == name.toUpperCase());
+    }));
+  }
+
+  function getFoodTypes() {
+    var temp = [];
+    for (var i=0; i<foodTrucks.length; i++) {
+      for (var j=0; j<foodTrucks[i].type.length; j++) {
+        temp.push(foodTrucks[i].type[j]);
+      }
+    }
+
+  //return unique food types only
+    return temp.filter(function(elem, index, self) {
+      return index == self.indexOf(elem);
+    }).sort();
+  }
+
+  function filterTrucksByFoodType(foodType) {
+    return foodTrucks.filter(function(item) {
+      for (var i=0; i<item.type.length; i++) {
+        if(item.type[i].toUpperCase() == foodType.toUpperCase()) {
+          return true;
+        }
+      }
+    });
+  }
+
+  //add new truck, input is a truck object
+  function addTruck(newTruck) {
+    foodTrucks.push(newTruck);
```

Comment:
This method needs to return the truck added. When you call this method in your post route, you're sending what this method returns to the response.